### PR TITLE
fix: Fix type errors in reducers

### PR DESCRIPTION
**Pull request by Gudahtt**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

All type errors in the reducers have been fixed. An error in the EngineState type was exposed during this process as well, and has also been fixed (the `PermissionControllerState` was resolving to `never` because the wrong type parameter was passed in).

The `fiatOrders` unit tests used selectors quite a lot, so there was a need for a mock/initial root state shape. This was introduced in the `app/util/test` directory for use in unit tests. This initial state object represents just the minimal initial state that has been typed so far. After we migrate all reducers to TypeScript and fix their types, this state object will become increasingly complete/correct.

**Issue**

This is a follow-up to #7110, which laid the groundwork for these type fixes but also introduced some of them accidentally.

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/Ramp/utils/index.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
   toTokenMinimalUnit,
 } from '../../../../util/number';
 import { getOrders, FiatOrder } from '../../../../reducers/fiatOrders';
-import { RootState } from '../../../../reducers/fiatOrders/types';
+import { RootState } from '../../../../reducers';
```
Comment (Gudahtt) [inline app/components/UI/Ramp/utils/index.ts:9]: This mistake was introduced in #7110, see https://github.com/MetaMask/metamask-mobile/pull/7110#discussion_r1316193739

File: app/core/Engine.ts [hunk 2]
```diff
@@ -133,7 +133,8 @@ type GlobalEvents =
   | PermissionControllerEvents
   | SignatureControllerEvents;
 
-type Permissions = ReturnType<typeof getPermissionSpecifications>;
+type PermissionsByRpcMethod = ReturnType<typeof getPermissionSpecifications>;
+type Permissions = PermissionsByRpcMethod[keyof PermissionsByRpcMethod];
```
Comment (Gudahtt) [inline app/core/Engine.ts:137]: The `Permissions` type parameter expected by `PermissionControllerState` is the type union of all of the permission objects. But the permission specification keys them by RPC method. This extracts the values of that mapping, getting just the permission objects directly, which is what the state type needs.

**Comment by Gudahtt** [comment]
Passing E2E smoke test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/5617a7ae-2bc8-4bb7-aac6-c80a31aa66a9

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**